### PR TITLE
drivers: serial: uart_altera: add driver

### PR DESCRIPTION
**Pull request by gohshunjing**
Add driver for altera avalon uart core.

Signed-off-by: Goh Shun Jing <shun.jing.goh@intel.com>

### CODE REVIEW

**Comment by dcpleung** [comment]
Please update copyright to 2023.

File: drivers/serial/uart_altera.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022 Intel Corporation
+ * Copyright (c) 2023 Intel Corporation
```
Comment (dcpleung) [inline drivers/serial/uart_altera.c:2]: Please squash this to the first commit.

**Review by dcpleung — CHANGES_REQUESTED** [review]

File: drivers/serial/uart_altera.c (hunk 1 above)
Comment (gohshunjing) [reply drivers/serial/uart_altera.c:2]: done

**Review by dcpleung — DISMISSED** [review]

**Review by dcpleung — DISMISSED** [review]

**Comment by nashif** [comment]
What platform/soc will be using this driver?

**Comment by gohshunjing** [comment]
Hello @nashif,
This driver is for [altera avalon uart soft ip](https://www.intel.com/content/www/us/en/docs/programmable/683130/22-3/uart-core.html) that can be included in designs with Nios2 or NiosV processors.
There are already plans to upstream NiosV to Zephyr.

**Comment by gohshunjing** [comment]
force push to rebase and resolve conflicts

**Review by dcpleung — APPROVED** [review]

**Comment by gohshunjing** [comment]
currently, there are no board/platform which has this driver enabled by default.
Is there any test, which i can include an overlay to build this driver?
Thank you.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**